src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -1070,25 +1070,38 @@ def get_enable_ultra_ssd(self) -> bool:
         # this parameter does not need validation
         return enable_ultra_ssd
 
+    # Mutable Fips now allows changes after create
     def get_enable_fips_image(self) -> bool:
         """Obtain the value of enable_fips_image, default value is False.
-
         :return: bool
         """
+
         # read the original value passed by the command
         enable_fips_image = self.raw_param.get("enable_fips_image", False)
-        # try to read the property value corresponding to the parameter from the `agentpool` object
-        if (
-            self.agentpool and
-            hasattr(self.agentpool, "enable_fips") and      # backward compatibility
-            self.agentpool.enable_fips is not None
-        ):
-            enable_fips_image = self.agentpool.enable_fips
+        # In create mode, try and read the property value corresponding to the parameter from the `agentpool` object
+        if self.decorator_mode == DecoratorMode.CREATE:
+            if (
+                self.agentpool and
+                hasattr(self.agentpool, "enable_fips") and      # backward compatibility
```

Comment:
make "enable_fips" a const?

Reply:
This code section isn't changed, and the string is not reused won't fix